apps/studio/components/interfaces/Auth/Hooks/CreateHookSheet.tsx

```diff
@@ -300,52 +328,107 @@ export const CreateHookSheet = ({
+                                if (
+                                  hook.method.schema !== '' &&
+                                  hook.method.functionName !== '' &&
+                                  hook.method.functionName !== value
+                                ) {
+                                  const revokeStatements = getRevokePermissionStatements(
+                                    hook.method.schema,
+                                    hook.method.functionName
+                                  )
+                                  setStatements([...revokeStatements])
+                                }
+                              }
+                              field.onChange(value)
+                            }}
+                            disabled={field.disabled}
+                          />
+                        </FormControl_Shadcn_>
+                      </FormItemLayout>
+                    )}
+                  />
+                </div>
+
+                <div className="w-full">
+                  <FormField_Shadcn_
+                    key="postgresValues.statements"
+                    control={form.control}
+                    name="postgresValues.statements"
+                    render={({ field }) => {
+                      const schemaName = values.postgresValues.schema
+                      const functionName = values.postgresValues.functionName
+                        ? `${schemaName}.${values.postgresValues.functionName}`
+                        : '<postgres-function>'
+                      const grantStatements = [
+                        `-- Grant supabase_auth_admin permission to execute the function\ngrant execute on function ${functionName} to supabase_auth_admin;`,
+                        `-- Grant usage to supabase_auth_admin\ngrant usage on schema ${schemaName} to supabase_auth_admin;`,
+                        `-- Ensure that the function used as the hook is not accessible from the Data API\nrevoke execute on function ${functionName} from authenticated, anon, public;`,
+                      ]
+                      field.value = statements.join('\n\n')
+                      return (
+                        <FormItemLayout
+                          className="px-4"
+                          label="Additional Permissions Applied"
+                          description="These permission changes will be applied on the postgres function configured as the hook."
+                        >
+                          <div className={cn('h-64')}>
+                            <FormControl_Shadcn_>
+                              <CodeEditor
+                                id="postgres-hook-editor"
+                                isReadOnly={true}
+                                language="pgsql"
+                                value={statements.concat(grantStatements).join('\n\n')}
+                                onInputChange={field.onChange}
```

Comment:
Can the user change the statements? No need for all this logic, just a simple `div` will be ok.

Reply:
nope it should be read-only, the user shouldn't be able to modify the statements

Reply:
also, make the div have the same padding as the rest of the form.